### PR TITLE
feat update cps file upload component

### PR DESCRIPTION
**Pull request by korel-san**

### CODE REVIEW

File: projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts [hunk 1]
```diff
@@ -2,20 +2,38 @@ import { Component } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { CpsFileUploadComponent } from 'cps-ui-kit';
 import { Observable, catchError, from, map, of } from 'rxjs';
+import {
+  CpsButtonToggleComponent,
+  CpsButtonToggleOption
+} from '../../../../../cps-ui-kit/src/lib/components/cps-button-toggle/cps-button-toggle.component';
+import { CpsIconComponent } from '../../../../../cps-ui-kit/src/lib/components/cps-icon/cps-icon.component';
+import componentData from '../../api-data/cps-file-upload.json';
```
Comment (fateeand) [inline projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts:10]: ambiguous import

File: projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts [hunk 2]
```diff
@@ -2,20 +2,38 @@ import { Component } from '@angular/core';
...
   host: { class: 'composition-page' }
 })
 export class FileUploadPageComponent {
   componentData = ComponentData;
+  fileUploadOptions: CpsButtonToggleOption[] = [
+    { label: 'Pictures', value: '.jpg' },
+    { label: 'Documents', value: '.pdf' },
+    { label: 'Pictures', value: '.png' }
+  ];
+  selectedFileUploadType: CpsButtonToggleOption = this.fileUploadOptions[0];
```
Comment (fateeand) [inline projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts:36]: eslint: Expected blank line between class members


```suggestion
fileUploadOptions: CpsButtonToggleOption[] = [
    { label: 'Pictures', value: '.jpg' },
    { label: 'Documents', value: '.pdf' },
    { label: 'Pictures', value: '.png' }
  ];
  
  selectedFileUploadType: CpsButtonToggleOption = this.fileUploadOptions[0];
```

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.html [hunk 3]
```diff
@@ -28,6 +29,9 @@
         extensionsStringAsterisks ? '(' + extensionsStringAsterisks + ')' : ''
       }}</span
     >
+    <div class="cps-file-upload-dropzone-content">
```
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.html:32]: I don't like the idea of providing the information about files using `ng-content`.
It is weird and unclear why it is provided differently from the other contents of the rectangle.

I would suggest solving this by using a prop `@Input() fileInfo = "";`
 
```html
@if (fileInfo) {
    <cps-icon color="calm" icon="info-circle" size="xsmall"></cps-icon>
    <span>{{fileInfo}}</span>
}
``` 
or alternatively if html format matters (to preserve breaks, etc.):
```html
<span [innerHTML]="fileInfo"></span>
```

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts [hunk 4]
```diff
@@ -124,6 +140,11 @@ export class CpsFileUploadComponent implements OnInit {
   removeUploadedFile() {
     const name = this.uploadedFile?.name ?? '';
     this.uploadedFile = undefined;
+
+    if (this.fileInput) {
+      this.fileInput.value = '';
```
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts:145]: why do we need this logic?

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts [hunk 5]
```diff
@@ -69,9 +75,19 @@ export class CpsFileUploadComponent implements OnInit {
   extensionsStringAsterisks = '';
 
   isProcessingFile = false;
+  fileInput?: HTMLInputElement;
```
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts:78]: what about viewchild?

File: projects/composition/src/app/pages/file-upload-page/file-upload-page.component.html [hunk 6]
```diff
@@ -1,5 +1,27 @@
...
+    [fileDesc]="selectedFileUploadType!.label || ''"
+    width="500"
+    [fileProcessingCallback]="processUploadedFile"
+    (fileUploadFailed)="onFileUploadFailed($event)"
+    (fileUploaded)="onFileUploaded($event)"
+    (uploadedFileRemoved)="onUploadedFileRemoved($event)">
+  </cps-file-upload>
+
+  <br/>
+  <hr/>
```
Comment (fateeand) [inline projects/composition/src/app/pages/file-upload-page/file-upload-page.component.html:23]: I would suggest using `cps-divider` instead

File: projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts [hunk 7]
```diff
@@ -2,20 +2,38 @@ import { Component } from '@angular/core';
...
   styleUrls: ['./file-upload-page.component.scss'],
   host: { class: 'composition-page' }
 })
 export class FileUploadPageComponent {
   componentData = ComponentData;
+  fileUploadOptions: CpsButtonToggleOption[] = [
+    { label: 'Pictures', value: '.jpg' },
+    { label: 'Documents', value: '.pdf' },
+    { label: 'Pictures', value: '.png' }
+  ];
```
Comment (fateeand) [inline projects/composition/src/app/pages/file-upload-page/file-upload-page.component.ts:35]:
```suggestion
  fileUploadOptions: CpsButtonToggleOption[] = [
    { label: 'JPG image', value: '.jpg' },
    { label: 'PDF document', value: '.pdf' },
    { label: 'PNG image', value: '.png' }
  ];
```

File: package-lock.json
Comment (fateeand) [inline package-lock.json:1]: to be rolled back

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts (hunk 4 above)
Comment (korel-san) [reply projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts]: no need, I had some weird behavior, tried to fix it but now it seems I can't reproduce

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts (hunk 5 above)
Comment (korel-san) [reply projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts]: no need, I removed it

File: projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts [hunk 8]
```diff
@@ -67,11 +80,22 @@ export class CpsFileUploadComponent implements OnInit {
...
 
   isProcessingFile = false;
 
   ngOnInit(): void {
-    this.width = convertSize(this.width);
+    this.updateExtensionsString();
+    this.cvtWidth = convertSize(this.width);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/components/cps-file-upload/cps-file-upload.component.ts:92]: Let's specify that the class implements `OnChanges` interface

**Review by fateeand — APPROVED** [review]

**Pull request merged**